ComponentKit/DataSources/CKCollectionViewTransactionalDataSource.mm

```diff
@@ -93,14 +94,56 @@ - (void)transactionalComponentDataSource:(CKTransactionalComponentDataSource *)d
                   didModifyPreviousState:(CKTransactionalComponentDataSourceState *)previousState
                        byApplyingChanges:(CKTransactionalComponentDataSourceAppliedChanges *)changes
 {
-  [_collectionView performBatchUpdates:^{
-    applyChangesToCollectionView(_collectionView, _attachController, [_componentDataSource state], changes);
-    // Detach all the component layouts for items being deleted
-    [self _detachComponentLayoutForRemovedItemsAtIndexPaths:[changes removedIndexPaths]
-                                                    inState:previousState];
-    // Update current state
-    _currentState = [_componentDataSource state];
-  } completion:NULL];
+  const BOOL hasMutations = (changes.removedIndexPaths.count ||
+                      changes.insertedIndexPaths.count ||
+                      changes.movedIndexPaths.count ||
+                      changes.insertedSections.count ||
+                      changes.removedSections.count);
+  const BOOL hasUpdates = !!changes.updatedIndexPaths.count;
+  
+  CKTransactionalComponentDataSourceState *state = [_componentDataSource state];
+  
+  if (hasUpdates && !hasMutations) {
+    CKComponentBoundsAnimation boundsAnimation = {};
+    for (NSIndexPath *indexPath in changes.updatedIndexPaths) {
+      boundsAnimation = [[state objectAtIndexPath:indexPath] boundsAnimation];
+      if (boundsAnimation.duration)
+        break;
+    }
+    
+    if (boundsAnimation.duration) {
+      id boundsAnimationContext = CKComponentBoundsAnimationPrepareForCollectionViewBatchUpdates(_collectionView);
+      [UIView performWithoutAnimation:^{
+        [_collectionView performBatchUpdates:^{
+          _currentState = state;
+        } completion:^(BOOL finished) {}];
```

Comment:
Since this is called twice could we put it in a block and just execute the block here or below, on line 123? Might be a bit more obvious. It's also a pattern I've seen used elsewhere.